streams/src/main/java/org/apache/kafka/streams/state/internals/KeyValueToTimestampedKeyValueByteStoreAdapter.java

```diff
@@ -126,15 +128,22 @@ public <R> QueryResult<R> query(
         final PositionBound positionBound,
         final QueryConfig config) {
 
-
         final long start = config.isCollectExecutionInfo() ? System.nanoTime() : -1L;
-        final QueryResult<R> result = store.query(query, positionBound, config);
+        QueryResult<R> result = store.query(query, positionBound, config);
+        Position position = result.getPosition();
+        if (result.isSuccess()) {
+            byte[] res = (byte[]) result.getResult();
+            byte[] res1 = convertToTimestampedFormat(res);
+            result = (QueryResult<R>) QueryResult.forResult(res1);
```

Comment:
We need to set the position on the newly created result. We should also copy the "execution information" (also wondering if we should add a new entry to it?) \cc @aliehsaeedii WDYT?

Maybe we can actually re-use `InternalQueryResultUtil.copyAndSubstituteDeserializedResult` ?

Reply:
ok